stubs/pycocotools/pycocotools/coco.pyi

```diff
@@ -0,0 +1,89 @@
+from _typeshed import Incomplete, Self
+from pathlib import Path
+from typing import Generic, TypeVar, overload
+from typing_extensions import Literal, TypeAlias, TypedDict
+
+from . import _EncodedRLE
+
+# TODO: Use numpy types when #5768 is resolved.
+# import numpy as np
+# import numpy.typing as npt
+
+PYTHON_VERSION: Incomplete
+_NDArray: TypeAlias = Incomplete
+
+class _Image(TypedDict):
+    id: int
+    width: int
+    height: int
+    file_name: str
+
+_TPolygonSegmentation: TypeAlias = list[list[float]]
+
+class _RLE(TypedDict):
+    size: list[int]
+    counts: list[int]
+
+class _Annotation(TypedDict):
+    id: int
+    image_id: int
+    category_id: int
+    segmentation: _TPolygonSegmentation | _RLE | _EncodedRLE
+    area: float
+    bbox: list[float]
+    iscrowd: int
+
+_TSeg = TypeVar("_TSeg", _TPolygonSegmentation, _RLE, _EncodedRLE)
+
+class _AnnotationG(TypedDict, Generic[_TSeg]):
+    id: int
+    image_id: int
+    category_id: int
+    segmentation: _TSeg
+    area: float
+    bbox: list[float]
+    iscrowd: int
+
+class _Category(TypedDict):
+    id: int
+    name: str
+    supercategory: str
+
+class _Dataset(TypedDict):
+    images: list[_Image]
+    annotations: list[_Annotation]
+    categories: list[_Category]
+
+class COCO:
+    anns: dict[int, _Annotation]
+    dataset: _Dataset
+    cats: dict[int, _Category]
+    imgs: dict[int, _Image]
+    imgToAnns: dict[int, list[_Annotation]]
+    catToImgs: dict[int, list[int]]
+    def __init__(self, annotation_file: str | Path | None = ...) -> None: ...
+    def createIndex(self) -> None: ...
+    def info(self) -> None: ...
+    def getAnnIds(
+        self, imgIds: list[int] | int = ..., catIds: list[int] | int = ..., areaRng: list[float] = ..., iscrowd: bool | None = ...
```

Comment:
It's generally best to avoid using `list[T]` for parameter annotations. It can lead to "annoying" errors from a type checker, due to the fact that `list` is invariant. In this case, for the first two parameters, the method will accept an `int`, or [any object that passes the `_isArrayLike` function](https://github.com/ppwwyyxx/cocoapi/blob/71e284ef862300e4319aacd523a64c7f24750178/PythonAPI/pycocotools/coco.py#L136-L137). `_isArrayLike` [checks for whether an object is iterable and sized](https://github.com/ppwwyyxx/cocoapi/blob/71e284ef862300e4319aacd523a64c7f24750178/PythonAPI/pycocotools/coco.py#L63-L64), so `Collection[int]` is a good fit here. (You'll need to import `Collection` from `collections.abc`.)

The third parameter must support indexing, so `Sequence` is a better fit for that one.

```suggestion
        self, imgIds: Collection[int] | int = ..., catIds: Collection[int] | int = ..., areaRng: Sequence[float] = ..., iscrowd: bool | None = ...
```